### ISSUE TITLE
Any workspace is stuck in "Starting"

### ISSUE THREAD

**Issue by hunter86bg** [eclipse/che#22804]
### Describe the bug

After deploying Che with 'chectl server:deploy --platform openshift' , no matter which workspace template is selected - it always ends up in a "Starting" phase:
![Screenshot from 2024-02-05 00-55-57](https://github.com/eclipse/che/assets/22281444/e3eb2e22-097a-4520-a627-a30da142e7fe)

### Che version

7.80@latest

### Steps to reproduce

chectl server:deploy --platform openshift
chectl dashboard:open
login as kubeadmin
Create workspace -> Select Bash or Go

### Expected behavior

The deployment to either fail or succeed.

### Runtime

OpenShift

### Installation method

chectl/latest

### Environment

Linux

### Eclipse Che Logs

```shell
Logs are available at https://drive.google.com/file/d/1q8mzysjp5huAy6VyfrOY1Aj3V984s56A/view?usp=sharing
```

**Comment by hunter86bg** [comment]
Any hint where to start from ?

**Comment by tolusha** [comment]
Hello @hunter86bg 

What is DevWorkspace version installed?
You can check it by running the command:

```
oc get sub devworkspace-operator -n openshift-operators -o jsonpath='{.status.installedCSV}'
```

Please share DW operator logs as well:
```
oc logs deploy/devworkspace-controller-manager -c devworkspace-controller  -n openshift-operators
```

**Comment by hunter86bg** [comment]
Version: ```devworkspace-operator.v0.25.0```

The DW logs are [here](https://drive.google.com/file/d/1VbvdpyXdyCgf7_RzDE5hC8GNt5HrsHpt/view?usp=sharing).

**Comment by tolusha** [comment]
Basically it means, that workspace SA `workspaceb48e5c70f5344d85-sa` has no Image PullSecrets configured [1]
[1] https://github.com/devfile/devworkspace-operator/blob/main/pkg/provision/workspace/pull_secret.go#L72-L76

**Comment by hunter86bg** [comment]
I guess I missed that step in the deployment. I used the cli approach to deploy it.

Can you point me to the relevant documentation page how to properly set it up ?

**Comment by ibuziuk** [comment]
@hunter86bg could you try to install it via OperatorHub instead? https://access.redhat.com/documentation/en-us/red_hat_openshift_dev_spaces/3.11/html/administration_guide/installing-devspaces#installing-devspaces-on-openshift-using-the-web-console

**Comment by hunter86bg** [comment]
I have added  pull secrets  for 'registry.connect.redhat.com' , 'registry.connect.redhat.com' and 'quay.io', uninstalled the operator (including https://github.com/devfile/devworkspace-operator/blob/main/docs/uninstall.md) and installed it again.
Seems that both 'devWorkspace Operator' and 'Eclipse Che' are installed in 'openshift-operators' namespace.

**Comment by hunter86bg** [comment]
Could it be related to the self-signed certificate of the OKD ?

**Comment by tolusha** [comment]
@hunter86bg 
Could you try to delete a user namespace and try again?

**Comment by hunter86bg** [comment]
I've already deleted the che-kube-che-<hash> and eclypse-che (I guess it was a leftover from previous attempts) and redeployed the 2 operators from the UI to openshift-operators namespace. Then I run chectl to finish the deployment.

I have a suspicion the problem is related to the routes but I don't have enough knowledge on the topic.

**Comment by hunter86bg** [comment]
Definitely something is wrong, as it starts with 'workspace.progressTimeout=15m' and a few seconds later it switches back to 5m. In the end it fails prior the first min have passed:
```
{"level":"info","ts":"2024-02-07T21:56:12Z","logger":"controllers.DevWorkspace","msg":"Reconciling Workspace","Request.Namespace":"che-kube-admin-che-ekgkzp","Request.Name":"bash","devworkspace_id":"","resolvedConfig":"workspace.progressTimeout=15m"}
{"level":"info","ts":"2024-02-07T21:56:12Z","logger":"controllers.DevWorkspace","msg":"Reconciling Workspace","Request.Namespace":"che-kube-admin-che-ekgkzp","Request.Name":"bash","devworkspace_id":"workspace19c47a77860c41fa","resolvedConfig":"workspace.progressTimeout=15m"}
{"level":"info","ts":"2024-02-07T21:56:12Z","logger":"controllers.DevWorkspace","msg":"Reconciling Workspace","Request.Namespace":"che-kube-admin-che-ekgkzp","Request.Name":"bash","devworkspace_id":"workspace19c47a77860c41fa","resolvedConfig":"workspace.progressTimeout=15m"}
{"level":"info","ts":"2024-02-07T21:56:14Z","logger":"controllers.DevWorkspace","msg":"Reconciling Workspace","Request.Namespace":"che-kube-admin-che-ekgkzp","Request.Name":"bash","devworkspace_id":"workspace19c47a77860c41fa","resolvedConfig":"workspace.progressTimeout=300s,workspace.containerSecurityContext is set"}
{"level":"info","ts":"2024-02-07T21:56:14Z","logger":"controllers.DevWorkspace","msg":"Reconciling Workspace","Request.Namespace":"che-kube-admin-che-ekgkzp","Request.Name":"bash","devworkspace_id":"workspace19c47a77860c41fa","resolvedConfig":"workspace.progressTimeout=300s,workspace.containerSecurityContext is set"}

<OUTPUT TRUNCATED>
{"level":"error","ts":"2024-02-07T21:56:32Z","msg":"Reconciler error","controller":"devworkspace","controllerGroup":"workspace.devfile.io","controllerKind":"DevWorkspace","DevWorkspace":{"name":"bash","namespace":"che-kube-admin-che-ekgkzp"},"namespace":"che-kube-admin-che-ekgkzp","name":"bash","reconcileID":"147bc7d5-3447-44eb-8fd4-2e3d09718f78","error":"Get \"https://eclipse-che.apps.okd4.lab.localdomain/kube-admin/bash/3100/healthz\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235"}
{"level":"info","ts":"2024-02-07T21:56:32Z","logger":"controllers.DevWorkspace","msg":"Reconciling Workspace","Request.Namespace":"che-kube-admin-che-ekgkzp","Request.Name":"bash","devworkspace_id":"workspace19c47a77860c41fa","resolvedConfig":"workspace.progressTimeout=300s,workspace.containerSecurityContext is set"}
{"level":"error","ts":"2024-02-07T21:56:33Z","msg":"Reconciler error","controller":"devworkspace","controllerGroup":"workspace.devfile.io","controllerKind":"DevWorkspace","DevWorkspace":{"name":"bash","namespace":"che-kube-admin-che-ekgkzp"},"namespace":"che-kube-admin-che-ekgkzp","name":"bash","reconcileID":"337df845-014c-4a52-9160-fb213e5d1dc2","error":"Get \"https://eclipse-che.apps.okd4.lab.localdomain/kube-admin/bash/3100/healthz\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235"}
```

**Comment by hunter86bg** [comment]
I've wiped everything and deployed Red Hat Openshift Dev Spaces with the dsc tool - the same behaviour is observed. Obviously there is something in OKD 4.15 that Che doesn't like it.

**Comment by hunter86bg** [comment]
This one looks strange(pod/devspaces-operator-5dfc7bf748-2hs7x):
```
2024-02-08T19:15:19Z	ERROR	Reconciler error	{"controller": "devworkspacerouting", "controllerGroup": "controller.devfile.io", "controllerKind": "DevWorkspaceRouting", "DevWorkspaceRouting": {"name":"routing-workspace63c33ea51adc46b7","namespace":"snikolov-devspaces"}, "namespace": "snikolov-devspaces", "name": "routing-workspace63c33ea51adc46b7", "reconcileID": "eaa908e0-a70f-40d3-8c86-6133add5ee2f", "error": "Operation cannot be fulfilled on devworkspaceroutings.controller.devfile.io \"routing-workspace63c33ea51adc46b7\": StorageError: invalid object, Code: 4, Key: /kubernetes.io/controller.devfile.io/devworkspaceroutings/snikolov-devspaces/routing-workspace63c33ea51adc46b7, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 3608c4b8-28c9-41ea-8d32-d820d5dd626d, UID in object meta: "}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/remote-source/devspaces-images-operator/app/devspaces-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/remote-source/devspaces-images-operator/app/devspaces-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/remote-source/devspaces-images-operator/app/devspaces-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:235
	```

**Comment by hunter86bg** [comment]
@tolusha , as we get a "Successfully reconciled." half a second later, do you think there is a workaround ?

**Comment by tolusha** [comment]
It is hard to say, I haven't met such an error before.
Maybe @AObuchow @amisevsk have some ideas.

**Comment by hunter86bg** [comment]
Seems the problem was solved after updating OKD to 4.15.0-0.okd-2024-02-10-035534 .
I've tested Python, Go and C++ workspaces and they work as expected (both with temporary and permanent storage).
